backend/routers/auth.py

```
@@ -0,0 +1,108 @@
+        email=user.email,
+        hashed_password=hashed_password,
+        full_name=user.full_name,
+        role=UserRole.USER # Enforce USER role
+    )
+    db.add(new_user)
+    db.commit()
+    db.refresh(new_user)
+    return new_user
+
+@router.post("/token", response_model=Token)
+def login_for_access_token(form_data: OAuth2PasswordRequestForm = Depends(), db: Session = Depends(get_db)):
```

Ruff (0.14.14)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)